### PR TITLE
Implemented Validation check for Serial Number and DeviceInfo

### PR DESCRIPTION
**Pull request by adarshan-intel**
1. Implemented Validation check for device serial Number and DeviceInfo 
2. Created Unit Test to validate the function on different possible strings ( All tests passing )

### CODE REVIEW

File: protocol/src/test/java/org/fidoalliance/fdo/protocol/ManufacturingInfoTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                "??? ??? ???",
+                "'",
+                "\"",
+                "''",
+                "\"\"",
+                "'\"'",
+                "\"''''\"'\"",
+                "\"'\"'\"''''\"",
+                "<foo val=�bar� />",
+                "<foo val=�bar� />",
```
Comment (DukeDavis12) [inline protocol/src/test/java/org/fidoalliance/fdo/protocol/ManufacturingInfoTest.java:61]: Can you remove the redundant test cases ?

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/serialization/ManufacturingInfoDeserializer.java [hunk 2]
```diff
@@ -38,8 +56,23 @@ public ManufacturingInfo deserialize(JsonParser jp, DeserializationContext ctxt)
...
     info.setKeyEnc(PublicKeyEncoding.fromNumber(node.get(index++).intValue()));
-    info.setSerialNumber(node.get(index++).textValue());
-    info.setDeviceInfo(node.get(index++).textValue());
+
+    String serialNumber = node.get(index++).textValue();
+
+    if (isValidString(serialNumber)) {
+      info.setSerialNumber(serialNumber);
+    } else {
+      throw new InvalidMessageException("Invalid Serial Number");
```
Comment (DukeDavis12) [inline protocol/src/main/java/org/fidoalliance/fdo/protocol/serialization/ManufacturingInfoDeserializer.java:65]: Make changes to ensure that CI is passing.

File: protocol/src/test/java/org/fidoalliance/fdo/protocol/ManufacturingInfoTest.java (hunk 1 above)
Comment (adarshan-intel) [reply protocol/src/test/java/org/fidoalliance/fdo/protocol/ManufacturingInfoTest.java]: https://github.com/fido-device-onboard/pri-fidoiot/pull/643/commits/193ccdca63c131aa9169d00ccb3d452e6d494ea1

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/serialization/ManufacturingInfoDeserializer.java [hunk 3]
```diff
@@ -28,6 +32,27 @@ public ManufacturingInfoDeserializer(Class<?> t) {
     super(t);
   }
```
[the remark this answers is not in the document]
Comment (adarshan-intel) [reply protocol/src/main/java/org/fidoalliance/fdo/protocol/serialization/ManufacturingInfoDeserializer.java:33]: image

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/serialization/ManufacturingInfoDeserializer.java (hunk 2 above)
Comment (adarshan-intel) [reply protocol/src/main/java/org/fidoalliance/fdo/protocol/serialization/ManufacturingInfoDeserializer.java:70]: Added javadoc for the funciton

**Review by DukeDavis12 — APPROVED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

**Review by DukeDavis12 — APPROVED** [review]

**Pull request merged**